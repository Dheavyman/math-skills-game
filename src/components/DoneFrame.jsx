import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  doneStatus: PropTypes.string,
  handleResetGame: PropTypes.func.isRequired,
};

const defaultProps = {
  doneStatus: null,
};

/**
 * Done frame component
 *
 * @param {object} props - Properties passed to component
 *
 * @returns {object} React element
 */
const DoneFrame = (props) => {
  const { doneStatus, handleResetGame } = props;

  return (
    <div className="text-center">
      <h2>{doneStatus}</h2>
      <br />
      <button
        type="button"
        className="btn btn-info"
        onClick={handleResetGame}
      >
        Play Again
      </button>
    </div>
  );
};

DoneFrame.propTypes = propTypes;
DoneFrame.defaultProps = defaultProps;

export default DoneFrame;
