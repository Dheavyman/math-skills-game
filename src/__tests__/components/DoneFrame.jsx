import React from 'react';
import TestRenderer from 'react-test-renderer';

import DoneFrame from '../../components/DoneFrame';

describe('Done frame component', () => {
  const props = {
    doneStatus: 'Game Over',
    handleResetGame: jest.fn()
  };
  it('should render component successfully', async () => {
    const renderer = TestRenderer.create(<DoneFrame {...props} />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
