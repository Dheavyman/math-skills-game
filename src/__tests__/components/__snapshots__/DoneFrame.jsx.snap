// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Done frame component should render component successfully 1`] = `
<div
  className="text-center"
>
  <h2>
    Game Over
  </h2>
  <br />
  <button
    className="btn btn-info"
    onClick={[MockFunction]}
    type="button"
  >
    Play Again
  </button>
</div>
`;
